### DISCUSSION TITLE
docker cask conflicts with kubernetes-cli formulae

### DISCUSSION THREAD

**Discussion by chunjiw** [Homebrew/discussions#4060] [Casks]
Hello community!

I have been using kubectl formulae on a Mac without issue. Lately I need to use docker but it seems like the docker formulae is only a client; I need to install docker cask to get docker engine. But then I see that docker cask conflicts with kubectl. Does that mean I cannot use both at the same time? What if I install Docker Desktop by downloading dmg file from docker website?

Thank you in advance!

**Comment by cho-m** [comment]
The conflict is due to Docker Desktop including `kubectl` already and it creates a symlink at `/usr/local/bin/kubectl`. So, on Intel macOS default Homebrew prefix (`/usr/local`), it can overwrite the `kubernetes-cli` formula symlink.

Some ideas:
- You could just install the Cask and use the `kubectl` from there.
- You could install both and prepend `PATH` with `$HOMEBREW_PREFIX/opt/kubernetes-cli/bin` so that it is picked up on your shell before the Cask's copy.

EDIT: Also, the symlinks are created by Docker app on startup, not by Homebrew. So, installing the DMG directly will also write into `/usr/local/bin`.
